tests/core/DependencyTrackerTest.py

```diff
@@ -53,3 +53,16 @@ def test_resolve(self):
 
         self.assertEqual(uut.resolve(30), set())
         self.assertEqual(uut.resolve(40), {20})
+
+    def test_all_dependencies_resolved(self):
+        uut = DependencyTracker()
+
+        self.assertTrue(uut.all_dependencies_resolved)
+
+        uut.add(0, 1)
+
+        self.assertFalse(uut.all_dependencies_resolved)
+
+        uut.resolve(0)
+
+        self.assertTrue(uut.all_dependencies_resolved)
```

Comment:
imo, for completeness, we should also add a test case here in which some of the dependencies are forcefully resolved